src/smc-webapp/project_settings.cjsx

```diff
@@ -434,9 +435,18 @@ HideDeletePanel = rclass
     toggle_delete_project: ->
         @actions('projects').toggle_delete_project(@props.project.get('project_id'))
```

Comment:
Please move all the analytics events that are here to actions.  If you put them here, it's a bunch of extra code that has nothing to do with the component **in the component**, which will make eventual unit testing we need to do on components more complicated.  Also, and more importantly, there could be *many* different ways that the toggle_delete_project action (say) gets called.  They should all generate an analytics event, and would if this code were in the project actions class.   Optimally, there would be only one line added to each action (not 4 lines of code), and the actual explicit calls to analytics_event would all be in a new typescript file (thus much more cleanly separating the code).

Reply:
Actually, maybe I'm wrong.  We would lose some information about what is happening with what I'm suggesting, and it's easy enough to mock analytics_event to be a no-op for the purposes of unit testing.  So never mind.